src/EventBus/Middleware/ReplayFlaggingMiddleware.php

```diff
@@ -14,11 +14,17 @@ final class ReplayFlaggingMiddleware implements EventBusMiddleware
 {
     private static bool $replaying = false;
 
+    /**
+     * Static so that if multiple instances of the middleware are created accidentally, they are all in sync.
+     */
     public static function startReplayMode(): void
```

Comment:
Would it also work to make the `ReplayFlaggingMiddleware` a singleton?

Reply:
That was the original approach with the `ReplayFlaggingEventBus` decorator, but that was actually bugged because it was created in an `extend()` _around_ a shared service (= a singleton in pimple). And the extend was always called again, so everytime the event bus was injected somewhere a new `ReplayFlaggingEventBus` decorator was created. I tested that by doing a `var_dump()` in the constructor of the `ReplayFlaggingEventBus` and then running the replay command, and it was being instantiated about 5 times.

So I figured it was better to bake the singleton-behavior into the class itself then depend on a correct bootstrapping, since we will also be moving to another service container later and I've seen similar bugs pop up when SAPI3 was ported to a different service container. (It's easy to use `add()` instead of `addShared()` or something similar by accident)

Reply:
\+ The static methods make it easy to access it from any part of the code (for example the replay command) without having to get it from the service container or event bus in some way